### ISSUE TITLE
Undefined array key "table_schema"

### ISSUE THREAD

**Issue by Zio3D** [phpmyadmin/phpmyadmin#19248]
I ran the query "SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = 'Sql201167_1' AND table_name = 'tasks_url_query' AND column_name = 'tsCreationDateTime'" and it gave me the following error:

```
Warning in ./libraries/classes/Display/Results.php#3023
Undefined array key "table_schema"

Backtrace

Results.php#2836: PhpMyAdmin\Display\Results->getSpecialLinkUrl(
array,
string 'tsCreationDateTime',
array,
string 'column_name',
)
Results.php#2560: PhpMyAdmin\Display\Results->getRowValues(
,
array,
integer 0,
boolean false,
array,
string '',
boolean false,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns',
array,
)
Results.php#4174: PhpMyAdmin\Display\Results->getTableBody(
,
array,
array,
array,
boolean false,
)
Sql.php#1356: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#1612: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
array,
boolean false,
integer 1,
integer 1,
NULL,
,
array,
)
Sql.php#1848: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'information_schema',
string 'columns',
NULL,
,
string './themes/pmahomme/img/',
integer 1,
integer 1,
NULL,
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
ImportController.php#816: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/table/structure',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Routing.php#186: PhpMyAdmin\Controllers\ImportController->index(array)
./../../index.php#50: PhpMyAdmin\Routing::callControllerForRoute(
string '/import',
,
,
)
Warning in ./libraries/classes/Display/Results.php#3023
Undefined array key "table_name"

Backtrace

Results.php#2836: PhpMyAdmin\Display\Results->getSpecialLinkUrl(
array,
string 'tsCreationDateTime',
array,
string 'column_name',
)
Results.php#2560: PhpMyAdmin\Display\Results->getRowValues(
,
array,
integer 0,
boolean false,
array,
string '',
boolean false,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns',
array,
)
Results.php#4174: PhpMyAdmin\Display\Results->getTableBody(
,
array,
array,
array,
boolean false,
)
Sql.php#1356: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#1612: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
array,
boolean false,
integer 1,
integer 1,
NULL,
,
array,
)
Sql.php#1848: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'information_schema',
string 'columns',
NULL,
,
string './themes/pmahomme/img/',
integer 1,
integer 1,
NULL,
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
ImportController.php#816: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/table/structure',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Routing.php#186: PhpMyAdmin\Controllers\ImportController->index(array)
./../../index.php#50: PhpMyAdmin\Routing::callControllerForRoute(
string '/import',
,
,
)
```

I tried to run this query again but now on the first try it doesn't give me any error,
clicking on "Update query" instead now I get:

```
Warning in ./libraries/classes/Display/Results.php#3023
Undefined array key "table_schema"

Backtrace

Results.php#2836: PhpMyAdmin\Display\Results->getSpecialLinkUrl(
array,
string 'tsCreationDateTime',
array,
string 'column_name',
)
Results.php#2560: PhpMyAdmin\Display\Results->getRowValues(
,
array,
integer 0,
boolean false,
array,
string '',
boolean false,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns',
array,
)
Results.php#4174: PhpMyAdmin\Display\Results->getTableBody(
,
array,
array,
array,
boolean false,
)
Sql.php#1356: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#1612: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
array,
boolean false,
integer 1,
integer 1,
NULL,
,
array,
)
Sql.php#1848: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'information_schema',
string 'columns',
NULL,
,
string './themes/pmahomme/img/',
integer 1,
integer 1,
NULL,
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Sql.php#1695: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/sql&amp;db=information_schema&amp;table=columns&amp;sql_query=SELECT+column_name%2C+data_type%2C+character_maximum_length%0D%0AFROM+information_schema.columns%0D%0AWHERE+table_schema+%3D+%27Sql201167_1%27%0D%0AAND+table_name+%3D+%27tasks_url_query%27%0D%0AAND+column_name+%3D+%27tsCreationDateTime%27',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
SqlController.php#219: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/sql&amp;db=information_schema&amp;table=columns&amp;sql_query=SELECT+column_name%2C+data_type%2C+character_maximum_length%0D%0AFROM+information_schema.columns%0D%0AWHERE+table_schema+%3D+%27Sql201167_1%27%0D%0AAND+table_name+%3D+%27tasks_url_query%27%0D%0AAND+column_name+%3D+%27tsCreationDateTime%27',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Routing.php#186: PhpMyAdmin\Controllers\SqlController->index(array)
./../../index.php#50: PhpMyAdmin\Routing::callControllerForRoute(
string '/sql',
,
,
)
Warning in ./libraries/classes/Display/Results.php#3023
Undefined array key "table_name"

Backtrace

Results.php#2836: PhpMyAdmin\Display\Results->getSpecialLinkUrl(
array,
string 'tsCreationDateTime',
array,
string 'column_name',
)
Results.php#2560: PhpMyAdmin\Display\Results->getRowValues(
,
array,
integer 0,
boolean false,
array,
string '',
boolean false,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns',
array,
)
Results.php#4174: PhpMyAdmin\Display\Results->getTableBody(
,
array,
array,
array,
boolean false,
)
Sql.php#1356: PhpMyAdmin\Display\Results->getTable(
,
array,
array,
boolean false,
)
Sql.php#1612: PhpMyAdmin\Sql->getHtmlForSqlQueryResultsTable(
,
string './themes/pmahomme/img/',
array,
boolean false,
integer 1,
integer 1,
NULL,
,
array,
)
Sql.php#1848: PhpMyAdmin\Sql->getQueryResponseForResultsReturned(
,
array,
string 'information_schema',
string 'columns',
NULL,
,
string './themes/pmahomme/img/',
integer 1,
integer 1,
NULL,
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Sql.php#1695: PhpMyAdmin\Sql->executeQueryAndGetQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/sql&amp;db=information_schema&amp;table=columns&amp;sql_query=SELECT+column_name%2C+data_type%2C+character_maximum_length%0D%0AFROM+information_schema.columns%0D%0AWHERE+table_schema+%3D+%27Sql201167_1%27%0D%0AAND+table_name+%3D+%27tasks_url_query%27%0D%0AAND+column_name+%3D+%27tsCreationDateTime%27',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
SqlController.php#219: PhpMyAdmin\Sql->executeQueryAndSendQueryResponse(
array,
boolean false,
string 'information_schema',
string 'columns',
NULL,
NULL,
NULL,
NULL,
NULL,
string 'index.php?route=/sql&amp;db=information_schema&amp;table=columns&amp;sql_query=SELECT+column_name%2C+data_type%2C+character_maximum_length%0D%0AFROM+information_schema.columns%0D%0AWHERE+table_schema+%3D+%27Sql201167_1%27%0D%0AAND+table_name+%3D+%27tasks_url_query%27%0D%0AAND+column_name+%3D+%27tsCreationDateTime%27',
string './themes/pmahomme/img/',
NULL,
NULL,
string 'SELECT column_name, data_type, character_maximum_length FROM information_schema.columns WHERE table_schema = \'Sql201167_1\' AND table_name = \'tasks_url_query\' AND column_name = \'tsCreationDateTime\'',
NULL,
)
Routing.php#186: PhpMyAdmin\Controllers\SqlController->index(array)
./../../index.php#50: PhpMyAdmin\Routing::callControllerForRoute(
string '/sql',
,
,
)
```

By running the query "SHOW CREATE TABLE `tasks_url_query`" I get:

```
CREATE TABLE `tasks_url_query` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `type` enum('query','url') NOT NULL,
  `url_or_query` text,
  `schedule_time` time DEFAULT NULL,
  `schedule_day` enum('daily','weekly','monthly') DEFAULT NULL,
  `last_executed` timestamp NULL DEFAULT NULL,
  `tsCreationDateTime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`)
) ENGINE=MyISAM AUTO_INCREMENT=2 DEFAULT CHARSET=utf8
```

### Screenshots

![Immagine-2024-08-09](https://github.com/user-attachments/assets/bb3dceeb-335b-4430-a001-ba0082532676)

### Server configuration

From Aruba phpMyAdmin:
    Server: 22.115.123.126 via TCP/IP
    Tipo di server: Percona Server
    Versione del server: 5.5.61-38.13-log - Percona Server (GPL), Release 38.13, Revision 812705b
    Versione protocollo: 10
    Utente: Sql201167@22.115.123.128
    Codifica caratteri del server: cp1252 West European (latin1)

### Client configuration

- Browser: Firefox 128.0.3 (64 bit)
- Operating system: Windows 10 22H2 19045.4717

### Additional context

Server IP, Database name, User IP and Username are not the original ones.

**Comment by liviuconcioiu** [comment]
What phpMyAdmin version Aruba uses? You didn't specified it.

From what I see, it might be a duplicate of #16791 which was fixed in 5.1.1.

**Comment by Zio3D** [comment]
Hello @liviuconcioiu,
I contacted Aruba customer service and they confirmed that I am using phpMyAdmin version 5.1.
Presumably it is the same error as bug #16791.
Thanks for the support.
